worlds/oot_soh/location_access/overworld/hyrule_field.py

```diff
@@ -5,8 +5,305 @@
+        (Locations.HF_SOUTHEAST_TREE_10, lambda bundle: (can_bonk_trees(bundle))),
+        (Locations.HF_SOUTHEAST_TREE_11, lambda bundle: (can_bonk_trees(bundle))),
+        (Locations.HF_SOUTHEAST_TREE_12, lambda bundle: (can_bonk_trees(bundle))),
+        (Locations.HF_SOUTHEAST_TREE_13, lambda bundle: (can_bonk_trees(bundle))),
+        (Locations.HF_SOUTHEAST_TREE_14, lambda bundle: (can_bonk_trees(bundle))),
+        (Locations.HF_SOUTHEAST_TREE_15, lambda bundle: (can_bonk_trees(bundle))),
+        (Locations.HF_SOUTHEAST_TREE_16, lambda bundle: (can_bonk_trees(bundle))),
+        (Locations.HF_SOUTHEAST_TREE_17, lambda bundle: (can_bonk_trees(bundle))),
+        (Locations.HF_SOUTHEAST_TREE_18, lambda bundle: (can_bonk_trees(bundle))),
+        (Locations.HF_SOUTHEAST_TREE_19, lambda bundle: (can_bonk_trees(bundle))),
+        (Locations.HF_CHILD_SOUTHEAST_TREE_1, lambda bundle: (is_child(bundle) and can_bonk_trees(bundle))),
+        (Locations.HF_CHILD_SOUTHEAST_TREE_2, lambda bundle: (is_child(bundle) and can_bonk_trees(bundle))),
+        (Locations.HF_CHILD_SOUTHEAST_TREE_3, lambda bundle: (is_child(bundle) and can_bonk_trees(bundle))),
+        (Locations.HF_CHILD_SOUTHEAST_TREE_4, lambda bundle: (is_child(bundle) and can_bonk_trees(bundle))),
+        (Locations.HF_CHILD_SOUTHEAST_TREE_5, lambda bundle: (is_child(bundle) and can_bonk_trees(bundle))),
+        (Locations.HF_CHILD_SOUTHEAST_TREE_6, lambda bundle: (is_child(bundle) and can_bonk_trees(bundle))),
+        (Locations.HF_TEKTITE_GROTTO_TREE, lambda bundle: (can_bonk_trees(bundle)))
+    ])
+    # Connections
+    connect_regions(Regions.HYRULE_FIELD, world, [
+        (Regions.LW_BRIDGE, lambda bundle: True),
+        (Regions.LAKE_HYLIA, lambda bundle: True),
+        (Regions.GERUDO_VALLEY, lambda bundle: True),
+        (Regions.MARKET_ENTRANCE, lambda bundle: True),
+        (Regions.KAKARIKO_VILLAGE, lambda bundle: True),
+        (Regions.ZR_FRONT, lambda bundle: True),
+        (Regions.LON_LON_RANCH, lambda bundle: True),
+        # Todo: Uses 'Here(RR_HYRULE_FIELD)'
+        (Regions.HF_SOUTHEAST_GROTTO, lambda bundle: (blast_or_smash(bundle))),
+        (Regions.HF_OPEN_GROTTO, lambda bundle: True),
+        (Regions.HF_INSIDE_FENCE_GROTTO, lambda bundle: (can_open_bomb_grotto(bundle))),
+        (Regions.HF_COW_GROTTO, lambda bundle: ((can_use(Items.MEGATON_HAMMER, bundle) or is_child(bundle)) and can_open_bomb_grotto(bundle))),
+        # Todo: Uses 'Here(RR_HYRULE_FIELD)'
+        (Regions.HF_NEAR_MARKET_GROTTO, lambda bundle: (blast_or_smash(bundle))),
+        # Todo: Uses 'Here(RR_HYRULE_FIELD)'
+        (Regions.HF_FAIRY_GROTTO, lambda bundle: (blast_or_smash(bundle))),
+        (Regions.HF_NEAR_KAK_GROTTO, lambda bundle: (can_open_bomb_grotto(bundle))),
+        (Regions.HF_TEKTITE_GROTTO, lambda bundle: (can_open_bomb_grotto(bundle))),
+    ])
+    
+    ## HF Southeast Grotto
+    # Locations
+    add_locations(Regions.HF_SOUTHEAST_GROTTO, world, [
```

Comment:
Your grottos are missing the grottoEvents. Check out `KF Storms Grotto` inside kokiri_forest.py for the events you need. Just give them enums that fit this region instead of course.

You have 3 grottos that need them. Make sure they all get unique EventLocations.